Documentation/Diagnostics/PH2112.md

```diff
@@ -0,0 +1,44 @@
+# PH2112: Avoid overridde with new keyword
+
+| Property | Value  |
+|--|--|
+| Package | [Philips.CodeAnalysis.MaintainabilityAnalyzers](https://www.nuget.org/packages/Philips.CodeAnalysis.MaintainabilityAnalyzers) |
+| Diagnostic ID | PH2112 |
+| Category  | [Maintainability](../Maintainability.md) |
+| Analyzer | [AvoidOverridingWithNewKeywordAnalyzer](https://github.com/philips-software/roslyn-analyzers/blob/master/Philips.CodeAnalysis.MaintainabilityAnalyzers/Maintainability/AvoidOverridingWithNewKeywordAnalyzer.cs)
+| CodeFix  | Yes |
+| Severity | Error |
+| Enabled By Default | Yes |
+
+## Introduction
+
+Overriding with the `new` keyword gives unexpected behavior for the callers of the overridden method or property.
+
+## How to solve
+
+See this article on [MS learn](https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/knowing-when-to-use-override-and-new-keywords).
+
+## Example
+
+Code that triggers a diagnostic:
+``` cs
+class BaseClass
+{
+    public int VirtualMethod() 
```

Comment:
I wouldn't call this "VirtualMethod". It's not virtual w/o the virtual keyword. Just call it "Method" or something like that.